src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -201,11 +201,7 @@ internal async Task<List<PostgresType>> LoadBackendTypes(NpgsqlConnection conn,
         {
             var commandTimeout = 0;  // Default to infinity
             if (timeout.IsSet)
-            {
                 commandTimeout = (int)timeout.TimeLeft.TotalSeconds;
-                if (commandTimeout <= 0)
```

Comment:
Now I think I remember why I didn't do this before... If the timeout has expired, and this returns 0, then the command below will have no timeout at all (since 0 means infinite).

So we may want to have an accessor that first checks the timeout - and throws TimeoutException if expired - and only if not expired, returns the time left (ensuring that it's never zero or negative). This way, code would be able to call it and know that everything is safe.

Does this make sense?

Reply:
And that's how it's done right now, isn't it?

Reply:
Ah yes, sorry - misread the code.